string.c

```diff
@@ -10450,33 +10450,59 @@ rb_str_unicode_normalized_p(int argc, VALUE *argv, VALUE str)
 /**********************************************************************
  * Document-class: Symbol
  *
- *  <code>Symbol</code> objects represent names and some strings
- *  inside the Ruby
- *  interpreter. They are generated using the <code>:name</code> and
- *  <code>:"string"</code> literals
- *  syntax, and by the various <code>to_sym</code> methods. The same
- *  <code>Symbol</code> object will be created for a given name or string
- *  for the duration of a program's execution, regardless of the context
- *  or meaning of that name. Thus if <code>Fred</code> is a constant in
- *  one context, a method in another, and a class in a third, the
- *  <code>Symbol</code> <code>:Fred</code> will be the same object in
- *  all three contexts.
- *
- *     module One
- *       class Fred
+ *  <code>Symbol</code> objects are often used like <code>String</code>
+ *  objects to represent names.
+ *
+ *  Symbols are generated using either of the following literal
+ *  syntaxes:
+ *
+ *  * <code>:name</code>
+ *  * <code>:"name with spaces"</code>
+ *
+ *  You can also convert some Ruby objects to symbols using common
+ *  <code>to_sym</code> conversion methods:
+ *
+ *     "some name".to_sym
+ *     => :"some name"
+ *
+ *  Symbols are often used instead of Strings because the same
+ *  <code>Symbol</code> object will be used in memory for a given
+ *  name or string for the duration of a program's execution.
+ *
+ *     class StringExample
+ *       def one
+ *         "important name"
+ *       end
+ *
+ *       def two
+ *         "important name"
  *       end
- *       $f1 = :Fred
- *     end
- *     module Two
- *       Fred = 1
- *       $f2 = :Fred
  *     end
- *     def Fred()
+ *
+ *     class SymbolExample
+ *       def one
+ *         :"important name"
+ *       end
+ *
+ *       def two
+ *         :"important name"
+ *       end
  *     end
- *     $f3 = :Fred
- *     $f1.object_id   #=> 2514190
- *     $f2.object_id   #=> 2514190
- *     $f3.object_id   #=> 2514190
+ *
+ *     StringExample.new.one.object_id   #=> 70308464584480
```

Comment:
Given Aaron's recent work on compaction, I don't think we should use object_id to describe memory addresses in documentation.

Reply:
In what situations does changing `object_id` break existing code? Sorry if I missed it.